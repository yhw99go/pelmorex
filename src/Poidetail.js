import React from 'react';

export default class Poidetail extends React.Component {
    constructor(props){ 
        super(props);
    }
    render(){
        return (
            <p>{this.props.location.state.label}</p>
        );
    }
}

