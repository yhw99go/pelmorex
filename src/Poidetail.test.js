import React from 'react';
import { shallow } from 'enzyme';
import Poidetail from './Poidetail.js';
import renderer from 'react-test-renderer'

describe('Poidetail', () => {

    it('matches snapshot', () => {
        const props = {
            location: {
                state: {
                    label : 'test label'
                }
            }
        };
        const wrapper = renderer.create(<Poidetail  {...props}/>).toJSON()
        expect(wrapper).toMatchSnapshot()
    });

    it('get proper label', () => {
        const props = {
            location: {
                state: {
                    label : 'test label'
                }
            }
        };
        const wrapper = shallow(<Poidetail {...props} />)
        expect(wrapper.text()).toContain("test label");
    });
});
