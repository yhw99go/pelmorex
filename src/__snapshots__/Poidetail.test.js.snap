// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Poidetail matches snapshot 1`] = `
<p>
  test label
</p>
`;
